openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -459,3 +465,7 @@ def _delete_cookie(response):
         secure=settings.SESSION_COOKIE_SECURE or None,
         httponly=settings.SESSION_COOKIE_HTTPONLY or None,
     )
+
+
+def _is_from_logout(request):
+    return hasattr(request, 'is_from_logout') and request.is_from_logout
```

Comment:
you can also pass a third param to `getattr`:

```python
def _is_from_logout(request):
    return getattr(request, 'is_from_logout', False)
```

calling `hasattr` isn't any faster, since it's implemented by calling `getattr` anyway: https://docs.python.org/2/library/functions.html#hasattr